llms/mlx_lm/utils.py

```diff
@@ -150,14 +186,21 @@ def generate(
         print("=" * 10)
         print("Prompt:", prompt)
 
-    prompt = mx.array(tokenizer.encode(prompt))
+    prompt_tokens = mx.array(tokenizer.encode(prompt))
+    repetition_context = []
+    repetition_context.extend([token.item() for token in prompt_tokens])
```

Comment:
nit: don't create a repetition context if the user doesn't apply the repetition penalty.